python/cugraph/cugraph/gnn/dgl_extensions/cugraph_service_store.py

```diff
@@ -0,0 +1,582 @@
+            The edge type to be added. This should follow the string format
+            '(src_type),(edge_type),(dst_type)'
+            If not specified, the type of properties will be added as
+            an empty string.
+        src_offset: int,
+            The offset to add for the source node type
+            defaults to zero
+        dst_offset: int,
+            The offset to add for the dst node type
+            defaults to zero
+        feat_name : string or dict {}
+            The feature name under which we should save the added properties
+            (ignored if contains_vector_features=False and the col names of
+            the dataframe are treated as corresponding feature names)
+        contains_vector_features : False
+            Whether to treat the columns of the dataframe being added as
+            as 2d features
+        Returns
+        -------
+        None
+        """
+        c_ar, len_ar = self.client.call_extension(
+            func_name="add_edge_data_from_parquet_remote",
+            file_path=file_path,
+            node_col_names=node_col_names,
+            canonical_etype=canonical_etype,
+            src_offset=src_offset,
+            dst_offset=dst_offset,
+            graph_id=self.gdata._graph_id,
+            result_device=self.device_id,
+        )
+        loaded_columns = _deserialize_strings_from_char_ars(c_ar, len_ar)
+        columns = [col for col in loaded_columns if col not in node_col_names]
+        _update_feature_map(
+            self.edata_feat_col_d, feat_name, contains_vector_features, columns
+        )
+        self.__clear_cached_properties()
+
+    def get_node_storage(self, key, ntype=None, indices_offset=0):
+        if ntype is None:
+            ntypes = self.ntypes
+            if len(self.ntypes) > 1:
+                raise ValueError(
+                    (
+                        "Node type name must be specified if there "
+                        "are more than one node types."
+                    )
+                )
+            ntype = ntypes[0]
+        if key not in self.ndata_feat_col_d:
+            raise ValueError(
+                f"key {key} not found in CuGraphStore node features",
+                f" {list(self.ndata_feat_col_d.keys())}",
+            )
+
+        columns = self.ndata_feat_col_d[key]
+        return CuFeatureStorage(
+            pg=self.gdata,
+            columns=columns,
+            storage_type="node",
+            indices_offset=indices_offset,
+            backend_lib=self.backend_lib,
+        )
+
+    def get_edge_storage(self, key, etype=None, indices_offset=0):
+        if etype is None:
+            etypes = self.etypes
+            if len(self.etypes) > 1:
+                raise ValueError(
+                    (
+                        "Edge type name must be specified if there "
+                        "are more than one edge types."
+                    )
+                )
+
+            etype = etypes[0]
```

Comment:
Where does `etype` get used after this line?

Reply:
Added tests here, similar to ntype. 
https://github.com/rapidsai/cugraph/blob/d24ab529b910687b586d358f62da3fe0b3b95783/python/cugraph/cugraph/tests/test_dgl_extension_graph_store.py#L413-L423